File: src/apps/Andriod/app/src/main/java/com/dkellycollins/triggerfinger/data/daos/ICollidableDao.java
package com.dkellycollins.triggerfinger.data.daos;

import com.dkellycollins.triggerfinger.data.entity.ICollidable;
import com.dkellycollins.triggerfinger.data.model.IVector;

/**
 * Provides the ability to manage collidable data.
 */
public interface ICollidableDao extends IActivityDao {

    Iterable<ICollidable> retrieve();
    ICollidable retrieve(int id);

    int create(IVector position, float radius);

    void update(int id, IVector position, float radius);

    void delete(int id);

}

File: src/apps/Android-Kotlin/app/src/main/kotlin/com/dkellycollins/triggerfinger/data/daos/ICollidableDao.kt
package com.dkellycollins.triggerfinger.data.daos

import com.dkellycollins.triggerfinger.data.entity.ICollidable
import com.dkellycollins.triggerfinger.data.model.IVector

/**
 * Provides the ability to manage collidable data.
 */
interface ICollidableDao : IActivityDao {

    fun retrieve(): Iterable<ICollidable>
    fun retrieve(id: Int): ICollidable

    fun create(position: IVector, radius: Float): Int

    fun update(id: Int, position: IVector, radius: Float)

    fun delete(id: Int)

}
